process/smartContract/scQueryService.go

```diff
@@ -88,6 +102,22 @@ func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice ui
 	log.Trace("executeScCall", "function", query.FuncName, "numQueries", service.numQueries)
 	service.numQueries++
 
+	shouldEarlyExitBecauseOfSyncState := query.ShouldBeSynced && service.bootstrapper.GetNodeState() == common.NsNotSynchronized
+	if shouldEarlyExitBecauseOfSyncState {
+		return nil, process.ErrNodeIsNotSynced
+	}
+
+	shouldCheckRootHashChanges := query.SameScState
+	rootHashBeforeExecution := make([]byte, 0)
+	var err error
+
+	if shouldCheckRootHashChanges {
+		rootHashBeforeExecution, err = service.accounts.RootHash()
```

Comment:
you do not need the accounts for this, you can use the blockchain's current header's roothash
Let's remove the extra accounts db pointer.

Reply:
done